packages/x-date-pickers/src/AdapterDateFns/index.ts

```diff
@@ -1 +1,53 @@
-export { default as AdapterDateFns } from '@date-io/date-fns';
+import BaseAdapterDateFns from '@date-io/date-fns';
+import defaultLocale from 'date-fns/locale/en-US';
+// @ts-ignore
+import longFormatters from 'date-fns/_lib/format/longFormatters';
+import { MuiFormatTokenMap, MuiPickerFieldAdapter } from '../internals/models';
+
+const formatTokenMap: MuiFormatTokenMap = {
+  y: 'year',
+  yy: 'year',
+  yyy: 'year',
+  yyyy: 'year',
+  MMMM: 'month',
+  MM: 'month',
+  dd: 'day',
```

Comment:
```suggestion
  dd: 'day',
  DD: 'day',
```
Without `DD` the basic demo with changed adapter breaks